src/components/homepage/calculator-preview.tsx

```diff
@@ -0,0 +1,367 @@
+import { useEffect, useState } from "react";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "../ui/input";
+import { Switch } from "../ui/switch";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { CALCULATOR_CONSTANTS } from "@/lib/calculatorConstants";
+import { Button } from "../ui/button";
+import { motion } from "framer-motion";
+
+const minWage = CALCULATOR_CONSTANTS.MIN_WAGE;
+const nonTaxableAmountOfTaxBasis =
+  CALCULATOR_CONSTANTS.NON_TAXABLE_AMOUNT_OF_TAX_BASIS;
+const incomeTax = CALCULATOR_CONSTANTS.INCOME_TAX;
+const employeeDeduction = Number(
+  CALCULATOR_CONSTANTS.DEDUCTIONS.reduce(
+    (sum, deduction) => sum + deduction.employeeContribution,
+    0
+  ).toFixed(3)
+);
+const companyDeduction = Number(
+  CALCULATOR_CONSTANTS.DEDUCTIONS.reduce(
+    (sum, deduction) => sum + deduction.employerContribution,
+    0
+  ).toFixed(3)
+);
+const maxOneChildrenUnderDeduction =
+  CALCULATOR_CONSTANTS.MAX_ONE_CHILDREN_UNDER_DEDUCTION;
+const maxOneChildrenOverDeduction =
+  CALCULATOR_CONSTANTS.MAX_ONE_CHILDREN_OVER_DEDUCTION;
+
+//total children rates based on numbers from form
+function calculateChildAllowanceRate(totalChildren: number): number {
```

Comment:
V 98% prípadoch nikdy neotypovávaj návratový typ z funkcie. Typescript si to vie odvodiť sám a väčšinou to odvodí ovela lepšie ako ty sám 😄 je lepšie si na to zvyknút a otypovávať návratový typ iba vtedy keď si si 100% istý že vieš čo robíš.